src/components/Viewport.ts

```diff
@@ -1178,15 +1193,26 @@ export default class Viewport {
     const state = this.state;
     const currentPanel = this.getCurrentPanel();
     const currentState = this.stateMachine.getState();
-    const isFreeScroll = (this.options.moveType as MoveTypeObjectOption).type === "freeScroll";
+    const isFreeScroll = this.moveType.is(MOVE_TYPE.FREE_SCROLL);
+    const relativeHangerPosition = this.getRelativeHangerPosition();
+    const halfGap = this.options.gap / 2;
 
-    if (currentState.holding || currentState.playing || isFreeScroll) {
+    if (currentState.holding || currentState.playing) {
       return;
     }
 
-    let newPosition = currentPanel
-      ? currentPanel.getAnchorPosition() - state.relativeHangerPosition
-      : this.getCameraPosition();
+    let newPosition: number;
+    if (isFreeScroll) {
+      const nearestPanel = this.getNearestPanel();
+
+      newPosition = nearestPanel
+        ? nearestPanel.getPosition() - halfGap + (nearestPanel.getSize() + 2 * halfGap) * state.panelMaintainRatio - relativeHangerPosition
```

Comment:
Could relativePosition be omitted if `panelMaintainRatio` is calculated by `cameraPosition` instead of `hangerPosition`?

Reply:
I think not. It can be same.

```js
nearestPanel.getPosition() - relativeHangerPosition - halfGap + (nearestPanel.getSize() + 2 * halfGap) * state.panelMaintainRatio
```